### DISCUSSION TITLE
New releases increases on semantic patch version

### DISCUSSION THREAD

**Discussion by iamNoah1** [danny-avila/LibreChat#5110] [Q&A] [answered]
I'm curious why new releases are only increasing the patch version, implying that they only fix something. Instead, they qualify as minor, maybe even major releases IMHO as they contain a lot of new features. Thus, I would expect this to be reflected in the semantic version number accordingly.

**Comment by danny-avila** [comment]
To be honest, I don't follow semantic versioning rules at all - I just increment the numbers somewhat arbitrarily. I probably should be more systematic about it, but for now I'm just bumping versions as I go without too much thought about what constitutes a major, minor or patch change.

To some effect I'm trying to keep versioning like this before a proper v1.0.0 where then I would follow it more closely.

This was started before I understood the convention and I kept it as I went along.

**Reply by samling** [reply, marked as answer]
In fairness to you, the [semver spec](https://semver.org/#spec-item-4) says anything 0.y.z is considered unstable and that anything can change at any time. Essentially, it’s within spec not to follow the spec until an official 1.0.0 release.

**Reply by iamNoah1** [reply]
Fair enough, thanks for the feedback :)
